src/visualizers/widgets/TextEditor/TextEditorWidget.js

```diff
@@ -1,79 +1,111 @@
-/*globals $, define*/
+/*globals $, define, monaco */
 /*jshint browser: true*/
 
 define([
-    'ace/ace',
     'underscore',
-    './completer',
     'js/Utils/ComponentSettings',
+    'text!./MonacoLanguages.json',
+    'vs/editor/editor.main',
     'jquery-contextMenu',
     'css!./styles/TextEditorWidget.css'
 ], function (
-    ace,
     _,
-    Completer,
-    ComponentSettings
+    ComponentSettings,
+    MonacoLanguages
 ) {
     'use strict';
 
-    var TextEditorWidget,
-        WIDGET_CLASS = 'text-editor',
+    const WIDGET_CLASS = 'text-editor',
         LINE_COMMENT = {
             python: '#',
             lua: '--',
             yaml: '#'
         };
 
-    TextEditorWidget = function (logger, container, config={}) {
-        this.logger = logger.fork('Widget');
+    MonacoLanguages = JSON.parse(MonacoLanguages);
+    const DEFAULT_THEMES = ['vs-dark', 'vs', 'hc-black'];
+    const DEFAULT_COLORS = {
+        'vs': '#fffffe',
+        'vs-dark': '#1e1e1e',
+        'hc-black': '#000000'
+    };
 
+    const AVAILABLE_KEYBINDINGS = ['default'];
```

Comment:
It looks like this can be removed.

Reply:
Do we still have support for vim keybindings?

Reply:
Yeah. That's the plan. https://github.com/brijeshb42/monaco-vim is the repository. It can be added as a node module/ as a library for the widget, wondering which is the best path to follow.

Reply:
We have support for vim keybindings now :)